File: src/main/java/ru/ioleksiv/telegram/bot/core/model/method/group/GetChatMembersCount.java
package ru.ioleksiv.telegram.bot.core.model.method.group;

import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.method.interfaces.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#getchatmemberscount">getChatMembersCount</a>
 */
public class GetChatMembersCount extends ChatAction<Integer> {
    private static final String METHOD = "getChatMembersCount";

    protected GetChatMembersCount(Networker networker) {
        super(METHOD, networker);
    }

    @Override
    public Class<? extends CommonResponse<Integer>> getResultWrapperClass() {
        return ResponseCollection.IntegerResponse.class;
    }

    @Override
    public GetChatMembersCount setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/group/GetChatMembersCount.kt
package io.github.oleksivio.tl.kbot.server.api.method.group

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.IntegerResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId

/**
 * @see [getChatMembersCount](https://core.telegram.org/bots/api/#getchatmemberscount)
 */
data class GetChatMembersCount(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    override val chatId: ChatId
) : ChatAction<Int>() {

    @JsonIgnore
    override val resultWrapperClass = IntegerResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getChatMembersCount"
}
